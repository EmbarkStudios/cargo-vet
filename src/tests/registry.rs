use super::*;

#[test]
fn test_registry_parse_error() {
    // Check that we can recover from an invalid registry with a useful error.
    let _enter = TEST_RUNTIME.enter();
    let mock = MockMetadata::simple();

    let metadata = mock.metadata();
    let (config, audits, imports) = builtin_files_no_exemptions(&metadata);

    let mut network = Network::new_mock();
    network.mock_serve(
        crate::storage::REGISTRY_URL,
        r#"
[registry.remote]
url = 10 # invalid!
"#,
    );

    let cfg = mock_cfg(&metadata);

    let store = Store::mock_online(&cfg, config, audits, imports, &network, false).unwrap();

    let report = crate::resolver::resolve(&metadata, None, &store);
    let suggest = report
        .compute_suggest(&cfg, &store, Some(&network))
        .unwrap();

    let human_output = BasicTestOutput::new();
    report
        .print_human(&human_output.clone().as_dyn(), &cfg, suggest.as_ref())
        .unwrap();

    insta::assert_snapshot!(human_output.to_string());
}
