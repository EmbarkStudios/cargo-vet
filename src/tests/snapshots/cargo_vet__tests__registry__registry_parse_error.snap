---
source: src/tests/registry.rs
expression: human_output.to_string()
---
Vetting Failed!

3 unvetted dependencies:
  third-party1:10.0.0 missing ["safe-to-deploy"]
  third-party2:10.0.0 missing ["safe-to-deploy"]
  transitive-third-party1:10.0.0 missing ["safe-to-deploy"]

recommended audits for safe-to-deploy:
    Command                                           Publisher  Used By       Audit Size
    cargo vet inspect third-party1 10.0.0             UNKNOWN    first-party   100 lines
    cargo vet inspect third-party2 10.0.0             UNKNOWN    first-party   100 lines
    cargo vet inspect transitive-third-party1 10.0.0  UNKNOWN    third-party1  100 lines

estimated audit backlog: 300 lines

WARNING: Import suggestions are disabled due to an incompatible registry. Consider upgrading to the most recent release of cargo-vet.

Use |cargo vet certify| to record the audits.

